src/api/procedures/utils.ts

```diff
@@ -216,6 +216,41 @@ export async function assertConfidentialAccountExists(account: ConfidentialAccou
   }
 }
 
+/**
+ * @hidden
+ * */
+export async function assertConfidentialAssetsEnabledForVenue(
+  venueId: BigNumber,
+  assets: ConfidentialAsset[]
+): Promise<void> {
+  const filterDetails = await Promise.all(
+    assets.map(async asset => {
```

Comment:
In the past, I have encountered a few scenarios where .map with async doesn't work as expected. Either we map the elements to a Promise array and then await on all or use a for-of loop instead of map and async. Not sure if this is working as expected.

Reply:
I agree its a bit of a tricky construct. 

> Either we map the elements to a Promise array and then await on all 

This is what is happening here. Adding `async` to the inner map function means the inferred return type gets wrapped in `Promise`. The `.map` executes immediately with no blocking and becomes Promise<{asset, details}>[].

`await Promise.all(` on the above line is key here.


`P.map` is from Bluebird, which they themselves recommend using native promise support where possible.

> If there is a feature that keeps you using bluebird. Please let us know so we can try and upstream it :)
Currently - it is only recommended to use Bluebird if you need to support old browsers or EoL Node.js or as an intermediate step to use warnings/monitoring to find bugs. 
https://github.com/petkaantonov/bluebird/?tab=readme-ov-file#%EF%B8%8Fnote%EF%B8%8F

Bluebird isn't deprecated exactly, so I don't think its worth removing at this point, but I think we should avoid using it, since it largely exists due to deficiencies with older versions of JS.